### PR TITLE
PT-6909 Extracted reader options to builder from private methods

### PR DESCRIPTION
**Pull request by vgarvardt**
One more piece extracted from https://github.com/hellofresh/klepto/pull/110

This time - reader options extracted from implementations private methods to a builder function.

### CODE REVIEW

File: pkg/reader/reader_test.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	assert.Equal(t, tableCfg.Filter.Limit, tableOpt.Limit)
+	assert.Equal(t, tableCfg.Filter.Sorts, tableOpt.Sorts)
+
+	require.Equal(t, len(tableCfg.Relationships), len(tableOpt.Relationships))
+	for i := range tableCfg.Relationships {
+		assert.Equal(t, tableCfg.Relationships[i].Table, tableOpt.Relationships[i].Table)
+		assert.Equal(t, tableCfg.Relationships[i].ForeignKey, tableOpt.Relationships[i].ForeignKey)
+		assert.Equal(t, tableCfg.Relationships[i].ReferencedTable, tableOpt.Relationships[i].ReferencedTable)
+		assert.Equal(t, tableCfg.Relationships[i].ReferencedKey, tableOpt.Relationships[i].ReferencedKey)
+	}
```
Comment (boekkooi-fresh) [inline pkg/reader/reader_test.go:46]: Why not use `assert.EqualValues`?

File: pkg/reader/reader.go [hunk 2]
```diff
@@ -74,7 +74,28 @@ type (
 	}
 )
 
-// Connect acts as fectory method that returns a reader from a DSN
+// NewReadTableOpt builds read table options from table config
+func NewReadTableOpt(tableCfg *config.Table) ReadTableOpt {
+	var rOpts []*RelationshipOpt
```
Comment (boekkooi-fresh) [inline pkg/reader/reader.go:79]:
```suggestion
	rOpts := make([]*RelationshipOpt, 0, len(tableCfg.Relationships))
```

File: pkg/dumper/query/dumper.go [hunk 3]
```diff
@@ -47,16 +47,13 @@ func (d *textDumper) Dump(done chan<- struct{}, spec *config.Spec, concurrency i
...
+		if tableConfig == nil {
 			log.WithField("table", tbl).Debug("no configuration found for table")
 		}
 
-		if table != nil {
-			opts = reader.ReadTableOpt{
-				Limit:         table.Filter.Limit,
-				Relationships: d.relationshipConfigToOptions(table.Relationships),
-			}
+		if tableConfig != nil {
```
Comment (aleksandrzhiliaev) [inline pkg/dumper/query/dumper.go:55]: since you are touching this part, maybe just add `else` here?

File: pkg/reader/reader_test.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+				ReferencedKey:   "r2-reference-key",
+			},
+		},
+	}
+
+	tableOpt := NewReadTableOpt(tableCfg)
+
+	assert.Equal(t, tableCfg.Filter.Match, tableOpt.Match)
+	assert.Equal(t, tableCfg.Filter.Limit, tableOpt.Limit)
+	assert.Equal(t, tableCfg.Filter.Sorts, tableOpt.Sorts)
```
Comment (aleksandrzhiliaev) [inline pkg/reader/reader_test.go:38]: should be here `require` as well?

File: pkg/reader/reader_test.go (hunk 1 above)
Comment (vgarvardt) [reply pkg/reader/reader_test.go:46]: `tableCfg.Relationships` and `tableOpt.Relationships` have different types, but fields list is the same

File: pkg/reader/reader_test.go (hunk 4 above)
Comment (vgarvardt) [reply pkg/reader/reader_test.go:38]: not need to stop test if this condition is failing, we can check other asserts as well even if this one is bad

File: pkg/reader/reader.go (hunk 2 above)
Comment (vgarvardt) [reply pkg/reader/reader.go]: done, but slightly different

File: pkg/dumper/query/dumper.go (hunk 3 above)
Comment (vgarvardt) [reply pkg/dumper/query/dumper.go]: done

**Comment by boekkooi-fresh** [comment]
:+1:

**Pull request merged**